src/Umbraco.Core/IO/FileSecurityException.cs

```diff
@@ -1,20 +1,46 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
+using System.Runtime.Serialization;
 
 namespace Umbraco.Core.IO
 {
+    /// <summary>
+    /// The exception that is thrown when the caller does not have the required permission to access a file.
+    /// </summary>
+    /// <seealso cref="System.Exception" />
+    [Obsolete("Throw an UnauthorizedAccessException instead.")]
```

Comment:
I've marked `FileSecurityException` obsolete, as it's only used once and just indicates unauthorized access (https://docs.microsoft.com/en-us/dotnet/api/system.unauthorizedaccessexception):
https://github.com/umbraco/Umbraco-CMS/blob/853087a75044b814df458457dc9a1f778cc89749/src/Umbraco.Core/IO/PhysicalFileSystem.cs#L317